apps/emqx/src/emqx_misc.erl

```diff
@@ -557,16 +557,47 @@ try_to_existing_atom(Convert, Data, Encoding) ->
         _:Reason -> {error, Reason}
     end.
 
-is_sensitive_key(token) -> true;
-is_sensitive_key("token") -> true;
-is_sensitive_key(<<"token">>) -> true;
-is_sensitive_key(password) -> true;
-is_sensitive_key("password") -> true;
-is_sensitive_key(<<"password">>) -> true;
-is_sensitive_key(secret) -> true;
-is_sensitive_key("secret") -> true;
-is_sensitive_key(<<"secret">>) -> true;
-is_sensitive_key(_) -> false.
+is_sensitive_key(token) ->
+    true;
+is_sensitive_key("token") ->
+    true;
+is_sensitive_key(<<"token">>) ->
+    true;
+is_sensitive_key(password) ->
+    true;
+is_sensitive_key("password") ->
+    true;
+is_sensitive_key(<<"password">>) ->
+    true;
+is_sensitive_key(secret) ->
+    true;
+is_sensitive_key("secret") ->
+    true;
+is_sensitive_key(<<"secret">>) ->
+    true;
+is_sensitive_key([{str, Bin}]) when is_binary(Bin) ->
+    is_sensitive_key(Bin);
+is_sensitive_key(Atom) when is_atom(Atom) ->
+    is_sensitive_key(erlang:atom_to_binary(Atom));
+is_sensitive_key(Bin) when is_binary(Bin), (size(Bin) =:= 19 orelse size(Bin) =:= 13) ->
+    try
+        %% This is wrapped in a try-catch since we don't know that Bin is a
+        %% valid string so string:lowercase/1 might throw an exception.
+        %%
+        %% We want to convert this to lowercase since the http header fields
+        %% are case insensitve, which means that a user of the webhook bridge
+        %% can write this field name in many different ways.
+        LowercaseBin = iolist_to_binary(string:lowercase(Bin)),
```

Comment:
Why is `iolist_to_binary` necessary?

Reply:
As we discussed `string:lowercase(Bin)` might always return a binary if the input is a Bin but this might change in the future if it is not documented.